### PR TITLE
Fill find_package2 bookkeeping for oneself as PRIVATE package

### PR DESCRIPTION
**Pull request by dennisklein**
Otherwise it will not be known in the dependency table summary.

### CODE REVIEW

File: src/modules/FairFindPackage2.cmake.in [hunk 1]
```diff
@@ -112,3 +112,8 @@ macro(find_package2 qualifier pkgname)
   set(CMAKE_PREFIX_PATH ${__old_cpp__})
   unset(__old_cpp__)
 endmacro()
+
+# Add oneself as PRIVATE package if used from the @PROJECT_NAME@ package
+if(@PROJECT_NAME@_FOUND)
+  set(PROJECT_PACKAGE_DEPENDENCIES ${PROJECT_PACKAGE_DEPENDENCIES} @PROJECT_NAME@)
```
Comment (ChristianTackeGSI) [inline src/modules/FairFindPackage2.cmake.in]: Why no `list(APPEND)`?

**Comment by ChristianTackeGSI** [comment]
What do you think of adding this to demo1?
```cmake
if (NOT "FairCMakeModules" IN_LIST PROJECT_PACKAGE_DEPENDENCIES)
  message(FATAL_ERROR "FairCMakeModules missing from PROJECT_PACKAGE_DEPENDENCIES!")
endif()
```

File: src/modules/FairFindPackage2.cmake.in (hunk 1 above)
Comment (dennisklein) [reply src/modules/FairFindPackage2.cmake.in]: Just copied from the function which also does not use list append (yet). one reason for not using list append in the past was that not all cmake versions had it. Regarding content changes we should have some thought also into using some version check in each module, so it is clearer for which cmake versions they are intended.

Comment (ChristianTackeGSI) [reply src/modules/FairFindPackage2.cmake.in]: `list(APPEND)` doesn't even have a notice in the cmake docs. So it's likely 3.0?

+1 on having version checks in each module. After all we just "install" them here, others just "use" them.

Comment (dennisklein) [reply src/modules/FairFindPackage2.cmake.in]: Added version check and `include_guard` as well as docs and test coverage for it.

> list(APPEND) doesn't even have a notice in the cmake docs. So it's likely 3.0?

Modernized to `list(APPEND ...)`.

**Comment by dennisklein** [comment]
Tests have more structure now, there is even more potential to refactor. Leaving it for now. Test coverage for `find_package2` is also not yet complete, but also leaving it for this PR.

File: cmake/TestLib.cmake [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+include_guard(GLOBAL)
+
+function(run_module_tests)
+  cmake_parse_arguments(PARSE_ARGV 0 ARGS "" "MODULE;PATH" "")
+
+  if(ARGS_PATH)
+    list(APPEND path "${ARGS_PATH}")
+  else()
+    list(APPEND path "${CMAKE_SOURCE_DIR}/src/modules")
```
Comment (ChristianTackeGSI) [inline cmake/TestLib.cmake]: You're using things like `head -20 "${path}/${ARGS_MODULE}.cmake"` below.

So please use `set()` here.

**Review by ChristianTackeGSI — APPROVED** [review]
One comment. Rest looks awesome!

File: cmake/TestLib.cmake (hunk 2 above)
Comment (dennisklein) [reply cmake/TestLib.cmake]: right, done

**Review by ChristianTackeGSI — APPROVED** [review]

**Pull request merged**